modules/boltlib/lib/puppet/functions/run_task.rb

```diff
@@ -59,21 +46,27 @@ def run_task_instance(task, *targets)
       )
     end
 
+    use_args = task_args.nil? ? {} : task_args
+    # TODO: Is it OK to refer to the Puppet::Pal namespace here?
+    Puppet::Pal::TaskSignature.new(task).runnable_with?(use_args) do |mismatch|
```

Comment:
Is it OK to refer the `Puppet::Pal` namespace from the implementation of the `run_task` puppet function provided by the `boltlib' module?

Reply:
There's a method name `#task_signature` on `Puppet::Pal::ScriptCompiler` which would allow you to retrieve a `TaskSignature` instance using a `Task` name. It requires you to get hold of the that compiler though and that's not possible just yet (this is what blocked me).

Reply:
In general, yes. Puppet::Pal is supposed to be the Puppets external API (PAL is short for Puppet As a Library).

Reply:
I actually used that method [in the bolt cli](https://github.com/puppetlabs/bolt/pull/190/files#diff-b4ccc59dee0e054966d665d51e08d67eR487).
I dint't use it here because:
- I didn't know how the get the hold of the compiler 😉
- I wanted to split the lookup and the parameter checking into two functions anyway